src/update-done/update-done.c

```diff
@@ -31,64 +28,23 @@
         "# was updated. See man:systemd-update-done.service(8).\n"
 
 static int apply_timestamp(const char *path, struct timespec *ts) {
-        struct timespec twice[2] = {
-                *ts,
-                *ts
-        };
-        _cleanup_fclose_ FILE *f = NULL;
-        int fd = -1;
+        _cleanup_free_ char *message = NULL;
         int r;
-        _cleanup_(unlink_and_freep) char *tmp = NULL;
-
-        assert(path);
-        assert(ts);
 
         /*
          * We store the timestamp both as mtime of the file and in the file itself,
          * to support filesystems which cannot store nanosecond-precision timestamps.
-         * Hence, don't bother updating the file, let's just rewrite it.
          */
 
-        r = mac_selinux_create_file_prepare(path, S_IFREG);
-        if (r < 0)
-                return log_error_errno(r, "Failed to set SELinux context for %s: %m", path);
-
-        fd = open_tmpfile_linkable(path, O_WRONLY|O_CLOEXEC, &tmp);
-        mac_selinux_create_file_clear();
-
-        if (fd < 0) {
-                if (errno == EROFS)
-                        return log_debug("Can't create temporary timestamp file %s, file system is read-only.", tmp);
-
-                return log_error_errno(errno, "Failed to create/open temporary timestamp file %s: %m", tmp);
-        }
-
-        f = fdopen(fd, "we");
-        if (!f) {
-                safe_close(fd);
-                return log_error_errno(errno, "Failed to fdopen() timestamp file %s: %m", tmp);
-        }
-
-        (void) fprintf(f,
-                       MESSAGE
-                       "TIMESTAMP_NSEC=" NSEC_FMT "\n",
-                       timespec_load_nsec(ts));
+        if (asprintf(&message,
+                     MESSAGE
+                     "TIMESTAMP_NSEC=" NSEC_FMT "\n",
+                     timespec_load_nsec(ts)) < 0)
+                return log_oom();
 
-        r = fflush_and_check(f);
+        r = write_string_file_atomic_label_ts(path, message, ts);
```

Comment:
this previously handled EROFS errors special and just debug logged about them. Do you knowingly drop that?